apps/web/src/components/WalletConnect/WcProvider.tsx

```diff
@@ -0,0 +1,82 @@
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { createWalletKit, useCurrentAccount, walletKit } from "@umami/state";
+import { formatJsonRpcError } from "@walletconnect/jsonrpc-utils";
+import { getSdkError, parseUri } from "@walletconnect/utils";
+import { type PropsWithChildren, useEffect } from "react";
+
+import SessionProposalModal from "./SessionProposalModal";
+
+export const context = {};
+
+export const WalletConnectProvider = ({ children }: PropsWithChildren) => {
+  const currentAccount = useCurrentAccount();
+  const { openWith } = useDynamicModalContext();
+
+  useEffect(() => {
+    const initializeWallet = async () => {
+      try {
+        await createWalletKit();
+        walletKit.on("session_proposal", proposal => void onSessionProposal(proposal));
+        walletKit.on("session_request", request => void onSessionRequest(request));
+      } catch (error) {
+        console.error("Error initializing Web3Wallet:", error);
+      }
+    };
+
+    void initializeWallet();
+  });
+
+  async function onSessionProposal(proposal: WalletKitTypes.SessionProposal) {
+    console.log("Session proposal received", proposal);
+    if (!currentAccount) {
```

Comment:
Why do we need `currentAccount` and this check? 
There will be account selector in `SessionProposalModal`

Also I assume this check will always be true anyway